pommerman/network/ion_server/match.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+"""IonServer Match handler - This contains functions responsible for playing matches (You shouldn't use this file directly)"""
+
+# Imports
+from multiprocessing import Pipe
+from .constants import SPCommands as spc
+from uuid import uuid4
+from os import listdir, makedirs
+from numpy import array
+from rapidjson import dumps, dump
+from gzip import compress
+from enum import Enum
+from numpy import ndarray, uint8, int64
+import pommerman
+
+
+# Functions
+def unique_uuid(dir):
+    """Generates a unique UUID and checks for collision"""
+    while (True):
+        uuid = str(uuid4())[:10]
+        try:
+            ls_dir = listdir(dir)
+        except FileNotFoundError:
+            makedirs(dir)
+            ls_dir = []
+        if not uuid + ".json" in ls_dir:
+            return uuid
+
+
+def resolve_classes(i):
+    """Resolves classes before JSON serialization"""
+    if (isinstance(i, tuple)):
+        i = list(i)
+    for key, value in enumerate(i):
+        if (isinstance(i, dict)):
```

Comment:
this is pretty late in the game to then ask about i being a dict.
you should consider rewriting this function entirely.

Reply:
The function makes total sense if you could write it in a better way I would recommend you to do so.